### ISSUE TITLE
Metis plugin: scan multiple network interfaces

### ISSUE THREAD

**Issue by hiperiondev** [f4exb/sdrangel#977]
I'm working in HPSDR interface for RPITX (https://github.com/hiperiondev/hpsdr-rpitx)
In particular I use a RPI W Zero with SPI ethernet card.
Sdrangel only recognize connection over wifi but not over ethernet.
Linhpsdr and Quisk works ok.

**Comment by f4exb** [comment]
> Sdrangel only recognize connection over wifi but not over ethernet.

Can you develop a bit more? Stated like that this does not make sense. WiFi or Ethernet do not make any difference on the network.

**Comment by hiperiondev** [comment]
I agree with you:  this does not make sense.
My test is this: start process in RPI waiting connection (connected over wifi and ethernet on differents networks)
From PC in the same situation (wifi and ethernet) I use  Linhpsdr and Quisk. In this case both programs retrieve the device in two places (net wifi and net ethernet). If I disconnect one I can discover on other connected network.
If I make same test with Sdrangel only in wifi can discover RPI.

**Comment by hiperiondev** [comment]
If you have a RPI can make same test with hpsdr simulator: https://github.com/hiperiondev/hpsdrsim

**Comment by f4exb** [comment]
It may well be possible that the broadcast packet is sent from just one network interface of the PC (it picks up the WiFi) and thus only devices on this network will reply. Although the broadcast is 255.255.255.255 it will not cross network boundaries.

What happens if you keep just the Ethernet interface up on the PC (i.e. shut down the WiFi)?

**Comment by hiperiondev** [comment]
> What happens if you keep just the Ethernet interface up on the PC (i.e. shut down the WiFi)?
Exactly the same thing happens

**Comment by f4exb** [comment]
Which means you don't see the RPi at all then?

**Comment by hiperiondev** [comment]
In RPI I have a service awaiting for a Metis discovery packet.
Over WIFI this packet arrives ok, but over ethernet never arrives.

With Linhpsdr and Quisk this does not happen

**Comment by f4exb** [comment]
OK it is still not clear sorry so let's be even more precise. Can you please shut down the WiFi completely and just use Ethernet. Then does the packet arrives through Ethernet or not?

**Comment by hiperiondev** [comment]
> OK it is still not clear sorry so let's be even more precise. Can you please shut down the WiFi completely and just use Ethernet. Then does the packet arrive through Ethernet or not?

That was the first test I did.
If I turn off the WIFI and leave only ethernet, the packets do not arrive

**Comment by f4exb** [comment]
So there is nothing to be done about it. I have a Red Pytaya on my Ethernet network that is perfectly recognized so you should check your networks settings instead.

**Comment by hiperiondev** [comment]
I'll try. Txs

**Comment by hiperiondev** [comment]
I reopen this issue after doing more testing.

Working on another project (https://github.com/hiperiondev/hpsdr-rtx-iq)) I ran into the same problem.

After searching a lot, I found that, when a broadcast is done (and only one) without associating the bind to an interface, it is Linux that chooses which one to exit from, and generally it does so according to the route table.

In particular, choose the interface that has the default route.
In my case it is the wifi, possibly in yours it is ethernet where the RedPitaya hardware is connected.

I tried to change my default route to ethernet and Sdrangel finds the device.

To be able to search in all the interfaces, a broadcat with bind must be sent to each interface, iterating in all the available ones.
